DBAAS-75 Add unit test for Atlas Operator inventory, connection and  dbaasprovider controllers

jianrongzhang89 opened the pull request:
This PR is to add unit test cases for Atlas Operator inventory, connection, and dbaasprovider controllers. Fake client and clientset are used, and a mock server for Atlas is also used. Issues found from unit test are also fixed.

make go-test-dbaas
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlasconnection	2.886s
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlasinventory	1.219s
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/dbaasprovider	1.939s

sgahlot reviewed (approved):
Tests look good to me but keep in mind that I haven't written any tests in Go.

jeremyary reviewed (commented):
Do we have or should we add a use case where it's a legitimate authentication, but there's no instances yet created to list? I just want to make sure we don't forget about the non-happy-path options as well. I can see a use case where evaluators set this up, maybe forget to or don't yet create any db's and start testing our integration.

jianrongzhang89 commented:
> Here's the output when I run the tests - are the ERROR outputs I'm seeing expected? Just wanted to check since there's no logger output in your example above. Thanks!
> 
> https://gist.github.com/jeremyary/e54e6cdbf0829ca31fb4a2df6f655ffd

I updated the test cases so that zaptest is used and it prevents the logs being mixed in the go test output (unless test fails). I also updated the test output above.

redhatHameed reviewed (commented):
@jianrongzhang89 added few comments. else /lgtm to me.

@jeremyary can we enable the GitHub action on this repo as workflow already configured WDYT?

jianrongzhang89 commented on Makefile:
```diff
@@ -211,3 +211,6 @@ catalog-build: opm ## Build a catalog image.
 catalog-push: ## Push a catalog image.
 	$(MAKE) docker-push IMG=$(CATALOG_IMG)
+
+go-test-dbaas: generate
```
It is not required and can be done through go test command directly:
CGO_ENABLED=0 go test ./pkg/controller/atlasconnection/... ./pkg/controller/atlasinventory/... ./pkg/controller/dbaasprovider/...

redhatHameed replied:
then we can keep it and add it into `make help`, as done for other commands.

jianrongzhang89 replied:
Sounds good. I added it back and added it into make help.

jianrongzhang89 commented on test.txt:
```diff
@@ -0,0 +1,54 @@
+/Users/zhangj/devt/src/mongodb-atlas-kubernetes/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
```
yes. it is removed.

redhatHameed reviewed (approved):
LGTM
